internal/servers/controller/handlers/groups/group_service_test.go

```diff
@@ -1356,3 +1359,68 @@ func TestRemoveMember(t *testing.T) {
 		})
 	}
 }
+
+// This is an analogue of the test in the scopeids package but pulled up to the
+// full service handler level.
+func TestRecursiveNoParentPerms(t *testing.T) {
+	require := require.New(t)
+	conn, _ := db.TestSetup(t, "postgres")
+	rw := db.New(conn)
+	wrap := db.TestWrapper(t)
+	kms := kms.TestKms(t, conn, wrap)
+	iamRepo := iam.TestRepo(t, conn, wrap)
+	iamRepoFn := func() (*iam.Repository, error) {
+		return iamRepo, nil
+	}
+	authTokenRepoFn := func() (*authtoken.Repository, error) {
+		return authtoken.NewRepository(rw, rw, kms)
+	}
+	serversRepoFn := func() (*servers.Repository, error) {
+		return servers.NewRepository(rw, rw, kms)
+	}
+	s, err := groups.NewService(iamRepoFn)
+	require.NoError(err)
+
+	o, p := iam.TestScopes(t, iamRepo)
+	at := authtoken.TestAuthToken(t, conn, kms, o.GetPublicId())
+	ctx := auth.NewVerifierContext(context.Background(),
+		nil,
+		iamRepoFn,
+		authTokenRepoFn,
+		serversRepoFn,
+		kms,
+		auth.RequestInfo{
+			Token:       at.GetToken(),
+			TokenFormat: auth.AuthTokenTypeBearer,
+			PublicId:    at.GetPublicId(),
+		})
+	// Create a group so that there is at least one value that will come back on
+	// success
+	g := iam.TestGroup(t, conn, p.GetPublicId())
+
+	// Attempt a recursive list at the org level with the given user. It has no
+	// permissions so there should overall be a Forbidden.
+	out, err := s.ListGroups(ctx, &pbs.ListGroupsRequest{
+		Recursive: true,
+		ScopeId:   o.GetPublicId(),
+	})
+	require.Error(err)
+	require.Nil(out)
+	require.Equal(err.Error(), handlers.ForbiddenError().Error())
+
+	// Add list on the project
+	r := iam.TestRole(t, conn, p.GetPublicId())
+	iam.TestRoleGrant(t, conn, r.GetPublicId(), "id=*;type=group;actions=list,read")
+	iam.TestUserRole(t, conn, r.GetPublicId(), "u_auth")
+
+	// Attempt a recursive list at the org level with the given user. It should
```

Comment:
Can you add 1 more case where you attempt a recursive list on global?

Along those same lines, there are resources that validate that list requests's scopes are for the scopes that the resource can be created in.  For example host_catalog_service.go validateListRequest validates the scope id as being a valid project scope.  There may be others.

Reply:
> Can you add 1 more case where you attempt a recursive list on global?

Actually I was likely going to remove this test as it's a much much much more stripped down version of what's in scope_ids_test.go. :-)

That's good feedback about validating list requests. I can skip that logic if it's recursive.

Reply:
Ah -- that's because it's not new here in this PR. Host catalogs already skips that validation if it's recursive! Same with targets.

Reply:
Removed this test in favor of the one that ended up in scope_ids_test.go.